### PR TITLE
Build multi-arch Operator pipeline image(ppc64le support)

### PR DESCRIPTION
**Pull request by mayurwaghmode**
Added GitHub Action for building multi-arch(x86 and ppc64le) image for Operator-pipeline-image.
Quay credentials need to be added as secret(QUAY_USER, QUAY_PWD) for pushing the multi-arch manifest to https://quay.io/repository/redhat-isv/operator-pipelines-images?tag=v1.0.7&tab=tags

### CODE REVIEW

**Comment by mayurwaghmode** [comment]
@MarcinGinszt  can you assign this PR for review?

**Comment by Allda** [comment]
Our Github CI is already configured to build and push image to the certain repository in quay.io. This change will break our CI because you replaced our namespace with yours. Can you please just update the existing build and push GitHub action to also build an extra architecture and push it to the existing repository?

**Comment by mayurwaghmode** [comment]
> Our Github CI is already configured to build and push image to the certain repository in quay.io. This change will break our CI because you replaced our namespace with yours. Can you please just update the existing build and push GitHub action to also build an extra architecture and push it to the existing repository?

Sure! I have updated the code. But I cannot push the built images to [this repo](quay.io/redhat-isv/operator-pipelines-images from my action) from my GitHub action as I don't have credentials of your repo.

**Comment by Allda** [comment]
We can test it for you since we have access to the namespace. However, you also removed a few other things like trigger events and image tags. Can you restore the previous functionality and just add the multiarch build?

**Comment by mayurwaghmode** [comment]
> We can test it for you since we have access to the namespace. However, you also removed a few other things like trigger events and image tags. Can you restore the previous functionality and just add the multiarch build?

Ok!

File: .github/workflows/main-image.yml [hunk 1]
```diff
@@ -21,26 +25,50 @@ jobs:
...
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y qemu-user-static
+
+      - name: Build image linux/amd64
+        id: build-image-amd64
         uses: redhat-actions/buildah-build@v2
         with:
           image: operator-pipelines-images
-          tags: latest ${{ github.sha }}
```
Comment (Allda) [inline .github/workflows/main-image.yml:29]: You remove a commit sha tag. Please revert the change.

File: .github/workflows/main-image.yml [hunk 2]
```diff
@@ -21,26 +25,50 @@ jobs:
...
           password: ${{ secrets.REGISTRY_PASSWORD }}
         if: ${{ github.event_name == 'push' }}
 
+      - name: Create and add to manifest
+        run: |
+          buildah manifest create ${{ env.IMAGE_NAME }}
+          buildah manifest add ${{ env.IMAGE_NAME }} ${{ env.IMAGE_NAME }}:linux-amd64
+          buildah manifest add ${{ env.IMAGE_NAME }} ${{ env.IMAGE_NAME }}:linux-ppc64le
+
+      - name: Push manifest to quay.io
```
Comment (Allda) [inline .github/workflows/main-image.yml:68]: Just curious.. why did you decide to remove push-to-quay step that uses `redhat-actions/push-to-registry@v2` and instead use a podman step?

Comment (mayurwaghmode) [reply .github/workflows/main-image.yml:68]: Actually, to build images for amd64 and ppc64le I am using two dockerfiles, so I am creating the manifest list and adding their manifests in one list. 
I faced error while doing it from redhat-actions/push-to-registry@v2 action

**Comment by mayurwaghmode** [comment]
All Changes are done!  @Allda  Can you review it once?

File: .github/workflows/main-image.yml [hunk 3]
```diff
@@ -20,13 +20,14 @@ jobs:
 
     steps:
       - uses: actions/checkout@v2
-
-      - name: Build Image
```
Comment (Allda) [inline .github/workflows/main-image.yml:24]: Please revert all changes in this file.

File: .github/workflows/ppc64le-image.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
+name: Build and Push Image
+on:  # yamllint disable-line rule:truthy
+  - push
+  - pull_request
+
+jobs:
+  test-lint:
```
Comment (Allda) [inline .github/workflows/ppc64le-image.yml:7]: This is already covered in `main-image.yml`. Please remove it

File: .github/workflows/ppc64le-image.yml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+          archs: ppc64le
+          dockerfiles: |
+            ./operator-pipeline-images/Dockerfile-ppc64le
+      - name: Push To quay.io
+        id: push-to-quay
+        uses: redhat-actions/push-to-registry@v2
+        with:
+          image: ${{ steps.build-image.outputs.image }}
+          tags: ${{ steps.build-image.outputs.tags }}
+          registry: quay.io/mayurwaghmode111
```
Comment (Allda) [inline .github/workflows/ppc64le-image.yml:44]: Update this line before merging the PR.

File: operator-pipeline-images/Dockerfile [hunk 6]
```diff
@@ -21,9 +21,13 @@ RUN dnf update -y && \
     dnf install -y \
     findutils \
     git \
+    wget \
```
Comment (Allda) [inline operator-pipeline-images/Dockerfile:24]: The original dockerfile should not be updated at all. Can you revert the changes.

File: operator-pipeline-images/Dockerfile-ppc64le [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    openssl-devel \
+    origin-clients \
+    pinentry \
+    pip \
+    python3-devel && \
+    dnf clean all
+
+COPY operator-pipeline-images/config/krb5.conf /etc/krb5.conf
+
+# Install opm CLI for x86
```
Comment (Allda) [inline operator-pipeline-images/Dockerfile-ppc64le:43]: How does this work if a Dockerfile is meant to be for ppc64le arch?

File: .github/workflows/ppc64le-image.yml [hunk 8]
```diff
@@ -0,0 +1,51 @@
+name: Build and Push Image
```
Comment (Allda) [inline .github/workflows/ppc64le-image.yml:1]: Can you please rename it to something else that is specific for extra arch?

Comment (mayurwaghmode) [reply .github/workflows/ppc64le-image.yml]: yes

File: .github/workflows/main-image.yml [hunk 9]
```diff
@@ -29,7 +29,7 @@ jobs:
           tags: latest ${{ github.sha }}
           dockerfiles: |
             ./operator-pipeline-images/Dockerfile
-
+
```
Comment (Allda) [inline .github/workflows/main-image.yml:32]: It still has some additional whitespaces.

**Review by Allda — APPROVED** [review]
